src/pyird/utils/irdstream.py

```diff
@@ -281,7 +302,13 @@ def calibrate_wavelength(self, trace_file_path=None, maxiter=30, stdlim=0.001, n
                 median_image, trace_legendre, y0, xmin, xmax, coeff)
             rsd = multiorder_to_rsd(rawspec, pixcoord)
             print(np.shape(rsd.T)[0])
-            wavsol, data = wavcal_thar(rsd.T, maxiter=maxiter, stdlim=stdlim)
+            ## set weights
+            if self.band=='h':
+                #w = make_weight()
+                w = np.ones(rsd.shape) #XXX!!
```

Comment:
memo:
One can set the weight of the fitting ThAr emissions in the wavelength calibration here.

Reply:
Yeah I don't know why the aperture size is related, hopefully this helps to resolve the inaccuracy in future PRs... 
I also tried using w from make_weight(); accuracy of the middle orders may have been slightly improved but those at the edges (i.e. 1,2 or 15,16) didn't change, so the origin might be different. Anyway thanks a lot for these updates!

Reply:
Thank you for trying to use make_weight. But it doesn't make significant improvement, I see...